### PR TITLE
wayland: fix cross

### PR DESCRIPTION
**Pull request by flokli**
meson needs to invoke nm from binutils during build

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please target staging.

File: pkgs/development/libraries/wayland/default.nix [hunk 1]
```diff
@@ -42,20 +57,32 @@ in stdenv.mkDerivation rec {
 
   depsBuildBuild = [
     pkgconfig
+    binutils
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/wayland/default.nix:60]: This is the diff is anyone is searching for it.

**Review by lukegb — APPROVED** [review]
Before:

```
$ nix build -f . pkgsCross.aarch64-multiplatform.wayland
error: build of '/nix/store/69j04pdvzrjxkf34wfkvc1bhdqkcdssg-wayland-1.18.0-aarch64-unknown-linux-gnu.drv' on 'ssh://whitby' failed: builder for '/nix/store/69j04pdvzrjxkf34wfkvc1bhdqkcdssg-wayland-1.18.0-aarch64-unknown-linux-gnu.drv' failed with exit code 1
builder for '/nix/store/69j04pdvzrjxkf34wfkvc1bhdqkcdssg-wayland-1.18.0-aarch64-unknown-linux-gnu.drv' failed with exit code 1; last 10 log lines:
  Build-time dependency wayland-scanner found: YES 1.18.0
  Program /nix/store/xmkd03p5i7pzjxqvsngyflmlhjl7dmxj-wayland-1.18.0/bin/wayland-scanner found: YES (/nix/store/xmkd03p5i7pzjxqvsngyflmlhjl7dmxj-wayland-1.18.0/bin/wayland-scanner)
  Library m found: YES
  Run-time dependency threads found: YES
  Program nm found: NO
  
  egl/meson.build:14:0: ERROR: Program 'nm' not found
  
  A full log can be found at /build/wayland-1.18.0/build/meson-logs/meson-log.txt
  builder for '/nix/store/69j04pdvzrjxkf34wfkvc1bhdqkcdssg-wayland-1.18.0-aarch64-unknown-linux-gnu.drv' failed with exit code 1
[3 built (1 failed), 58 copied (977.7 MiB), 194.5 MiB DL]
error: build of '/nix/store/69j04pdvzrjxkf34wfkvc1bhdqkcdssg-wayland-1.18.0-aarch64-unknown-linux-gnu.drv' failed
```

After:
```
$ nix build -f . pkgsCross.aarch64-multiplatform.wayland
[4 built, 34 copied (837.4 MiB), 163.2 MiB DL]
```

**Comment by Ericson2314** [comment]
See `nm` being mentioned in the `postPatch`? I think the issue is that it's not doing anything anymore. We should probably convert this from a sed to a patch so we'll catch this in the future.

File: pkgs/development/libraries/wayland/default.nix [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+, stdenv
+, fetchurl
+, fetchpatch
+, meson
+, pkgconfig
+, substituteAll
+, ninja
+, libffi
+, libxml2
+, wayland
```
Comment (primeos) [inline pkgs/development/libraries/wayland/default.nix:11]: Personally, I don't like this style but I get the reasons.

File: pkgs/development/libraries/wayland/default.nix [hunk 3]
```diff
@@ -26,6 +41,10 @@ in stdenv.mkDerivation rec {
       url = "https://gitlab.freedesktop.org/wayland/wayland/-/commit/e53e0edf0f892670f3e8c5dd527b3bb22335d32d.patch";
       sha256 = "15sbhi86m9k72lsj56p7zr20ph2b0y4svl639snsbafn2ir1zdb2";
     })
+    (substituteAll {
+      src = ./0001-add-placeholder-for-nm.patch;
+      nm = "${stdenv.cc.targetPrefix}nm";
+    })
```
Comment (primeos) [inline pkgs/development/libraries/wayland/default.nix:47]: That's neat :) I didn't know we had that function.

**Review by primeos — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**